tet/security/authentication.py

```diff
@@ -238,6 +238,40 @@ class CookieAttributes:
 MAX_PASSWORD_LENGTH = 128
 MIN_SCORE = 2
 KEY_PREFIX_PROFILE_CHANGE_PASSWORD_FORM = "settings.profile.changePasswordForm"
+MFA_REQUIRED_KEY = "mfa_required"
+
+
+@dataclasses.dataclass
+class AuthLoginFailure:
+    """
+    Dataclass for storing login failure information.
+
+    Attributes:
+        message: The error message describing the failure.
+        status_code: The HTTP status code associated with the failure.
+    """
+
+    message: str
+    status_code: int
+
+
+@dataclasses.dataclass
+class AuthLoginResult:
```

Comment:
Ad a field `success` to clearly indicate if the login was a success or not.

Also add `__bool__` method that uses this field so that you can something like
```
result = self.login_callback(request)
if result:
    ...
```